### PR TITLE
Add tests for letter imports

### PR DESCRIPTION
**Pull request by rz-comp**
Add several .tif files and code that tests each of them. For every file,
the filename and expected imported data is provided in a .csv index.

TestTools has been adapted to new templates.

### CODE REVIEW

File: sbc_compassion/__openerp__.py [hunk 1]
```diff
@@ -55,8 +55,8 @@
     ],
     'demo': [
         'demo/correspondence_template_demo.xml',
-        'demo/update_demo_partner_compassion.xml',
-        'demo/update_demo_child_compassion.xml',
+        # 'demo/update_demo_partner_compassion.xml',
+        # 'demo/update_demo_child_compassion.xml',
```
Comment (loikki) [inline sbc_compassion/__openerp__.py]: Should not be commented

File: sbc_compassion/tests/test_letter_import.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+from openerp.tests import common
+
+import base64
+import csv
+import os
+import re
+
+IMPORT_DIR = os.path.join(os.path.dirname(__file__), "testdata/import")
+
+def get_file_content(path):
```
Comment (loikki) [inline sbc_compassion/tests/test_letter_import.py]: Maybe this function can be moved to sbc_compassion/tools/import_letters_func.py

File: sbc_compassion/tests/test_letter_import.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    """ Reads a file and returns a base64 encoding of its contents. """
+    with open(path, "rb") as f:
+        file_content = f.read()
+        return base64.b64encode(file_content)
+
+def read_csv():
```
Comment (loikki) [inline sbc_compassion/tests/test_letter_import.py]: same as get_file_content

File: sbc_compassion/tests/test_letter_import.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        reader = csv.reader(csvfile)
+        header = reader.next()
+        for row in reader:
+            values = {}
+            for key, value in zip(header, row):
+                values[key] = value
+            file_list.append(values)
+    return file_list
+
+def import_letters(letters_obj, filename):
```
Comment (loikki) [inline sbc_compassion/tests/test_letter_import.py]: import_letterS but at the end, you just import one letter

**Comment by loikki** [comment]
You should try to explain the parameters in the docstring.
Usually in python, sphinx is used to create a documentation, therefore I recommend to use their style

File: sbc_compassion/tests/test_letter_import.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    file_content = get_file_content(path)
+    attachment = {
+        "name": filename,
+        "db_datas": file_content,
+    }
+    letter = letters_obj.create({"data": [(0, 0, attachment)]})
+    letter.button_import()
+    return letter
+
+def get_template_list(data_obj):
```
Comment (loikki) [inline sbc_compassion/tests/test_letter_import.py]: Can be moved

**Comment by loikki** [comment]
We need at least one zip in order to check the code that decompress them

File: sbc_compassion/tests/test_letter_import.py (hunk 3 above)
Comment (rz-comp) [inline sbc_compassion/tests/test_letter_import.py]: These functions are quite specific to tests in this module, so I think they should stay where they are

File: sbc_compassion/tests/test_letter_import.py (hunk 4 above)
Comment (rz-comp) [inline sbc_compassion/tests/test_letter_import.py]: There can be more than one letter in a file, updated doc to make this clear.

File: sbc_compassion/demo/update_demo_child_compassion.xml [hunk 6]
```diff
@@ -2,58 +2,72 @@
   <data>
     <record id="child_compassion.brazil" model="compassion.country">
       <field name="spoken_langs_ids" eval="[(4,lang_compassion_portuguese)]"/>
+      <field name="iso_code">BR</field>
```
Comment (loikki) [inline sbc_compassion/demo/update_demo_child_compassion.xml]: Already exists in child_compassion/demo

File: sbc_compassion/demo/update_demo_partner_compassion.xml [hunk 7]
```diff
@@ -3,272 +3,338 @@
     <record id="partner_compassion.partner_01" model="res.partner">
       <field name="mandatory_review">True</field>
       <field name="spoken_langs_ids" eval="[(4,lang_compassion_spanish)]"/>
+      <field name="name">Partner</field>
```
Comment (loikki) [inline sbc_compassion/demo/update_demo_partner_compassion.xml]: Strange that you need this, everything was alright on my computer

File: sbc_compassion/tests/test_letter_import.py [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+
+import base64
+import csv
+import os
+import re
+
+IMPORT_DIR = os.path.join(os.path.dirname(__file__), "testdata/import")
+
+
+def get_file_content(path):
```
Comment (loikki) [inline sbc_compassion/tests/test_letter_import.py]: Strange github has erased my previous comment about moving this function in tools/import_letters_functions.py

File: sbc_compassion/tests/test_letter_import.py [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+    """ Reads a file and returns a base64 encoding of its contents. """
+    with open(path, "rb") as f:
+        file_content = f.read()
+        return base64.b64encode(file_content)
+
+
+def read_csv():
```
Comment (loikki) [inline sbc_compassion/tests/test_letter_import.py]: same

File: sbc_compassion/tests/test_letter_import.py [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+        header = reader.next()
+        for row in reader:
+            values = {}
+            for key, value in zip(header, row):
+                values[key] = value
+            file_list.append(values)
+    return file_list
+
+
+def import_letters(letters_obj, filename):
```
Comment (loikki) [inline sbc_compassion/tests/test_letter_import.py]: same

File: sbc_compassion/tests/test_letter_import.py [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+    attachment = {
+        "name": filename,
+        "db_datas": file_content,
+    }
+    letters = letters_obj.create({"data": [(0, 0, attachment)]})
+    letters.button_import()
+    return letters
+
+
+def get_template_list(data_obj):
```
Comment (loikki) [inline sbc_compassion/tests/test_letter_import.py]: same

File: sbc_compassion/tests/test_letter_import.py [hunk 12]
```diff
@@ -0,0 +1,104 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2015 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Emanuel Cino <ecino@compassion.ch>
```
Comment (ecino) [inline sbc_compassion/tests/test_letter_import.py]: Please put your name

Comment (rz-comp) [inline sbc_compassion/tests/test_letter_import.py]: done

**Comment by ecino** [comment]
:+1:

**Pull request merged**